cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -682,7 +675,7 @@ public Location call() throws Exception {
           }
 
 
-          ImpersonationUtils.doAs(ugi, new Callable<Void>() {
+          impersonator.doAs(streamId, new Callable<Void>() {
```

Comment:
Anyone allowed to create a stream will need write privileges on the streams dir. The .deleted dir should have the same privileges as streams because stream owners should be able to delete their streams.